test/replay/btcstaking_test.go

```diff
@@ -0,0 +1,89 @@
+package replay
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/babylonlabs-io/babylon/testutil/datagen"
+	bstypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzCreatingAndActivatingDelegations(f *testing.F) {
```

Comment:
Seems we can extend it to supporting post-approval?

Reply:
yup, we can :) though first I want to work towards having votes from finality provider so we have full flow, and afterwards focus on adding different scenarios (pre-approval, post-approval, jailing, no-timestamped randomnes, slashing, unbonding...)